src/status_im/common/biometric/events.cljs

```diff
@@ -1,120 +1,128 @@
+    [status-im.common.biometric.utils :as utils]
     [status-im.common.keychain.events :as keychain]
     [status-im.constants :as constants]
     [taoensso.timbre :as log]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
-(def android-device-blacklisted?
-  (and platform/android? (= (:brand (native-module/get-device-model-info)) "bannedbrand")))
-
-(defn get-label-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-android (i18n/label :t/biometric-fingerprint)
-    constants/biometrics-type-face-id (i18n/label :t/biometric-faceid)
-    (i18n/label :t/biometric-touchid)))
-
-(defn get-icon-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-face-id :i/face-id
-    :i/touch-id))
-
-(re-frame/reg-fx
- :biometric/get-supported-biometric-type
+(rf/reg-fx
+ :biometric/get-supported-type
  (fn []
    ;;NOTE: if we can't save user password, we can't use biometric
    (keychain/can-save-user-password?
     (fn [can-save?]
-      (when (and can-save? (not android-device-blacklisted?))
+      (when (and can-save? (not utils/android-device-blacklisted?))
         (-> (biometrics/get-supported-type)
             (.then (fn [type]
-                     (rf/dispatch [:biometric/get-supported-biometric-type-success type])))))))))
+                     (rf/dispatch [:biometric/set-supported-type type])))))))))
+
+(defn set-supported-type
+  [{:keys [db]} [supported-type]]
+  {:db (assoc-in db [:biometrics :supported-type] supported-type)})
 
-(rf/defn get-supported-biometric-auth-success
-  {:events [:biometric/get-supported-biometric-type-success]}
-  [{:keys [db]} supported-type]
-  {:db (assoc db :biometric/supported-type supported-type)})
+(schema/=> set-supported-type events-schema/?set-supported-type)
+(rf/reg-event-fx :biometric/set-supported-type set-supported-type)
 
-(rf/defn show-message
-  {:events [:biometric/show-message]}
-  [_ error]
-  (let [code    (ex-cause error)
-        content (if (#{::biometrics/not-enrolled
-                       ::biometrics/not-available}
+(defn show-message
+  [_ [code]]
+  (let [content (if (#{:biometrics/not-enrolled-error
+                       :biometrics/not-available-error}
                      code)
                   (i18n/label :t/grant-face-id-permissions)
                   (i18n/label :t/biometric-auth-error {:code code}))]
-    {:effects.utils/show-popup
-     {:title   (i18n/label :t/biometric-auth-login-error-title)
-      :content content}}))
+    {:fx [[:effects.utils/show-popup
+           {:title   (i18n/label :t/biometric-auth-login-error-title)
+            :content content}]]}))
 
+(schema/=> show-message events-schema/?show-message)
+(rf/reg-event-fx :biometric/show-message show-message)
 
-(re-frame/reg-fx
+(rf/reg-fx
  :biometric/authenticate
- (fn [{:keys [on-success on-fail prompt-message]}]
+ (fn [{:keys [prompt-message on-success on-fail on-cancel on-done]
+       :or   {on-done    identity
```

Comment:
maybe it's cleaner to just wrap the functions below in `(when (fn? <> )` checks below?

Reply:
we no longer check for whether it's a function or vector. Reverted back to just using functions. Using :or here is to avoid all the repeated `(when on-done (on-done))` and making it clear that they're optional.